### DISCUSSION TITLE
Postgres: display generated sql before execute the query

### DISCUSSION THREAD

**Discussion by gjed** [grafana/grafana#50522] [Dashboards, Panels & Visualizations]
## What would you like to be added
That the generated sql of a panel query is generated and/or shown before the execution against the database, not after

## Why is this needed
It is needed because many times it is really helpful to debug the query with external tools, even just psql from the command line. 
The problem here is that if the query does't execute correctly, the generated sql it is not shown.

## Other information 
If it is helpful, I'm not using thw sql query build, but writing it directly

**Comment by gabor** [comment]
hi, in the current version of Grafana, i recommend opening the `Query Inspector`, and in it's `Query` tab you can see the exact query that was sent to the database. this works also when there was an error.

for example, here is a screenshot that happens when there is a syntax error in the query:
<img width="585" alt="sql" src="https://github.com/grafana/grafana/assets/51989/5edf968e-2ba6-4c38-a49e-4c9a9192bed6">

i hope this solves the problem.

if it does not solve the problem, please open a new github issue, and describe why the current solution is not sufficient. thanks!

**Reply by gjed** [reply]
That solves it! Thank you
